### PR TITLE
Simple command to list feeds

### PR DESCRIPTION
**Pull request by cableman**
#### Link to ticket

https://jira.itkdev.dk/browse/EV-288

#### Description

Simple command to list feeds currently in the database (with option to list disabled feeds as well).

#### Screenshot of the result

Screenshot from 2023-09-06 16-04-52

#### Checklist

- [ ] My code is covered by test cases.
- [x] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

Got tired of having to lookup ID's in the database after fixture changes and load.

### CODE REVIEW

File: src/Command/FeedListCommand.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        private readonly FeedConfigurationMapper $configurationMapper,
+        private readonly FeedRepository $feedRepository,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        // @todo: Add option to limit based on organization (data available in other PR currently).
+        $this->addOption('show-disabled', '', InputOption::VALUE_NONE, 'Also list disabled feeds');
```
Comment (rimi-itk) [inline src/Command/FeedListCommand.php:31]: `include-disabled` may be a better option name. Setting the flag will show _all_ feeds (including disabled ones) and not just disabled feeds.

Adding a `status` option with values `enabled` (default), `disabled`,  and `all`, say, might be a better approach.

File: src/Command/FeedListCommand.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $showDisabled = $input->getOption('show-disabled');
+
+        if ($showDisabled) {
+            $feeds = $this->feedRepository->findAll();
+        } else {
+            $feeds = $this->feedRepository->findBy(['enabled' => !$showDisabled]);
+        }
```
Comment (rimi-itk) [inline src/Command/FeedListCommand.php:46]:
```suggestion
        $feeds = $showDisabled
          ? $this->feedRepository->findAll()
          : $this->feedRepository->findBy(['enabled' => true]);
```

File: src/Command/FeedListCommand.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+        foreach ($feeds as $feed) {
+            $config = $this->configurationMapper->getConfigurationFromArray($feed->getConfiguration());
+            $io->definitionList(
+                ['Id' => $feed->getId()],
+                ['Name' => $feed->getName()],
+                ['Url' => $config->url]
+            );
```
Comment (rimi-itk) [inline src/Command/FeedListCommand.php:54]: Showing if the feed is disabled or not may be handy. Maybe only when `show-disabled` is set.

File: src/Command/FeedListCommand.php (hunk 1 above)
Comment (cableman) [reply src/Command/FeedListCommand.php:31]: Do you known if Symfony command supports an "select list", so it automatically only accepts these value?

File: src/Command/FeedListCommand.php (hunk 3 above)
Comment (cableman) [reply src/Command/FeedListCommand.php]: Fixed

File: src/Command/FeedListCommand.php (hunk 1 above)
Comment (cableman) [reply src/Command/FeedListCommand.php]: Fixed

File: src/Command/FeedListCommand.php [hunk 4]
```diff
@@ -37,20 +39,26 @@ protected function configure(): void
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $io = new SymfonyStyle($input, $output);
-        $showDisabled = $input->getOption('show-disabled');
+        $status = $input->getOption('status');
+        if (!in_array($status, self::SELECTIONS)) {
+            $io->error(sprintf('Status given not valid. Allowed values are (%s)', implode(', ', self::SELECTIONS)));
```
Comment (rimi-itk) [inline src/Command/FeedListCommand.php:44]: Rather that repeating the `implode`-stuff here, you could render the command help:

```suggestion
            $io->error(sprintf('Invalid status: %s', $status));

            // Show how to run this command.
            // https://symfony.com/doc/current/console/calling_commands.html
            $this->getApplication()->doRun(new ArrayInput([
                'command' => $this->getName(),
                '--help'  => true,
            ]), $output);
```

**Review by rimi-itk — APPROVED** [review]
Approved with a suggestion.

**Pull request merged**